### PR TITLE
(998) Simplify integration tests

### PR DESCRIPTION
**Pull request by yndajas**
## Context

We have a lot of unDRY code in our integration tests, which is making navigation within specs and comparison across contexts and specs a little painful. When adding new specs and contexts there's a not insignificant careful copy/paste overhead, which also rears its head in review

## Changes in this PR

- Simplify assignment and stubbing
- DRY up sign in

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

**Review by Gweaton — APPROVED** [review]

File: integration_tests/e2e/refer/programmeHistory.cy.ts [hunk 1]
```diff
@@ -503,97 +488,73 @@ context('Programme history', () => {
...
-      cy.task('stubCourse', course)
-
-      cy.signIn()
       cy.visit(path)
     })
 
     it('shows the delete page', () => {
       const deleteProgrammeHistoryPage = Page.verifyOnPage(DeleteProgrammeHistoryPage, {
-        participationWithName: courseParticipationWithName,
+        participationWithName: courseParticipationWithCourseIdWithName,
```
Comment (Gweaton) [inline integration_tests/e2e/refer/programmeHistory.cy.ts:529]: `courseIdAndName`? (sorry!)

**Review by Gweaton — APPROVED** [review]
A lot to take in, but it looks neat to me!

File: integration_tests/e2e/refer/programmeHistory.cy.ts (hunk 1 above)
Comment (yndajas) [reply integration_tests/e2e/refer/programmeHistory.cy.ts:529]: Ha, I thought that too but didn't do it for some reason

**Comment by yndajas** [comment]
[Simplify `check your answers` page integration tests to reuse setup between tests (S)](https://trello.com/c/JgbGz5KQ/998-simplify-check-your-answers-page-integration-tests-to-reuse-setup-between-tests-s)

**Pull request merged**